src/dmd/globals.d

```diff
@@ -718,4 +722,50 @@ enum PINLINE : ubyte
 
 alias StorageClass = uinteger_t;
 
+/// Support for `__DATE__`, `__TIME__`, and `__TIMESTAMP__`
+struct TimeStampInfo
+{
+    // Note: Those properties need to be guarded by a call to `init`
+    // The API isn't safe, and quite brittle, but it was left this way
+    // over performance concerns.
+    // This is currently only called once, from the lexer.
+    __gshared char[11 + 1] date;
+    __gshared char[8 + 1] time;
+    __gshared char[24 + 1] timestamp;
+
+    private static void initialize() nothrow
+    {
+        import core.stdc.time;
+        import core.stdc.stdio;
+        import core.stdc.stdlib;
+        import dmd.errors;
+        import dmd.utils;
+        import dmd.root.string;
+
+        time_t ct;
+        // https://issues.dlang.org/show_bug.cgi?id=20444
+        if (auto p = getenv("SOURCE_DATE_EPOCH"))
+        {
+            if (!ct.parseDigits(p.toDString()))
+                error(Loc.initial, "Value of environment variable `SOURCE_DATE_EPOCH` should be a valid UNIX timestamp, not: `%s`", p);
+        }
+        else
+            core.stdc.time.time(&ct);
+        const p = ctime(&ct);
+        assert(p);
+        sprintf(&date[0], "%.6s %.4s", p + 4, p + 20);
+        sprintf(&time[0], "%.8s", p + 11);
+        sprintf(&timestamp[0], "%.24s", p);
+    }
+
+    private static void deinitialize() nothrow
+    {
+        date = date.init;
+        time = time.init;
+        timestamp = timestamp.init;
+    }
```

Comment:
We don't need `deinitialize`, there's no memory to free, nor any interface to conform to.

Reply:
It stabilizes the state.
Rather than leaving stale data, there is no drawback to having it is there?

Reply:
What is unstable about the state ? If there was some actual cleanup / deallocation happening I would understand, but those are value types being set to `\xFF`.

> there is no drawback to having it is there?

IOW, there is no benefit in having it.

Reply:
I agree with @Geod24 - there's nothing to deinitialize.

Reply:
The benefit is that there in no stale data left in the timestamp after calling de-initialize.
Which is the whole point of having a deinitalize method in the first place.

Reply:
I see this as beneficial only for debugging purposes. Assigning on deinitialization is useless since the usage of uninitialized/deinitialized data is not supposed to happen.